### ISSUE TITLE
memory usage increasing

### ISSUE THREAD

**Issue by yueyihua** [vectr-ucla/direct_lidar_inertial_odometry#39]

**Comment by kennyjchen** [comment]
Yes, memory will increase with more keyframes placed in the environment.

**Comment by baptisteLynx** [comment]
Hello ! 
First thanks for the package, I have great results and the localization is really robust! 

I have the same “problem” as yueyihua, with the memory increasing as the mission goes on. I'm on Orin Nx, Ubuntu 20.04, Ros Noetic.

I was wondering if it was possible to set up some kind of buffer that could save and reset keyframes as they go (like a snapshot) and that would allow the memory used to be reduced regularly?

It would make the code even more lightweight!

Thanks again !
